lib/functions/modem.js

```diff
@@ -163,7 +163,7 @@ module.exports = function(SerialPort) {
         resultData.data = self.arrangeMessages(resultData.data)
         modem.emit('onNewMessageIndicator', resultData)
         modem.emit('onNewMessage', resultData.data[resultData.data.length - 1])
-        if (modem.autoDeleteOnReceive) {
+        if (modem.autoDeleteOnReceive && typeof message != 'undefined' ) {
```

Comment:
```suggestion
        if (modem.autoDeleteOnReceive && typeof message !== 'undefined' ) {
```

Reply:
äääähhmm .... where does "message" come from? or am I blind? There is no message initialized in this block ... ?!?